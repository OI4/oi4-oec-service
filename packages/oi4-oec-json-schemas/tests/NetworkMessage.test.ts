import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/NetworkMessage.schema.json'
import dataSetMessage from '../src/schemas/DataSetMessage.schema.json'
import uint16 from '../src/schemas/dataTypes/uint16.schema.json'
import uint32 from '../src/schemas/dataTypes/uint32.schema.json'
import configurationVersionDataType from '../src/schemas/ConfigurationVersionDataType.schema.json'

import validObjs from './__fixtures__/NetworkMessage_valid.json'
import invalidObjs from './__fixtures__/NetworkMessage_invalid.json'


expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [dataSetMessage, uint16, uint32, configurationVersionDataType],
    verbose: true,
  })
)

describe('NetworkMessage schema', () => {
    it('validate schema', () => {
      expect(schema).toBeValidSchema()
    })
  
    it.each(validObjs as [])(
      '(%#) match valid config object to schema -> %s',
      (_name: string, obj) => {
        expect(obj).toMatchSchema(schema)
      }
    )
  
    it.each(invalidObjs as [])(
      '(%#) match fails for invalid config -> %s',
      (name: string, obj) => {
        expect(obj).not.toMatchSchema(schema)
        expect(() => {
          expect(obj).toMatchSchema(schema)
        }).toThrowErrorMatchingSnapshot(name)
      }
    )
  })