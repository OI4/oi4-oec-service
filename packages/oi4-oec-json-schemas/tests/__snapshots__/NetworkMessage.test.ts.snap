// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkMessage schema (0) match fails for invalid config -> empty: empty 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'MessageId'
    Path:     NetworkMessage.schema.json#/required
  must have required property 'MessageType'
    Path:     NetworkMessage.schema.json#/required
  must have required property 'PublisherId'
    Path:     NetworkMessage.schema.json#/required
  must have required property 'Messages'
    Path:     NetworkMessage.schema.json#/required
"
`;

exports[`NetworkMessage schema (1) match fails for invalid config -> Network message with unsupported property: Network message with unsupported property 1`] = `
"expect(received).toMatchSchema(schema)

received
  must NOT have additional properties, but found 'unsupported'
    Expected: <boolean> false
    Path:     NetworkMessage.schema.json#/additionalProperties
"
`;

exports[`NetworkMessage schema (2) match fails for invalid config -> Network message without MessageId: Network message without MessageId 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'MessageId'
    Path:     NetworkMessage.schema.json#/required
"
`;

exports[`NetworkMessage schema (3) match fails for invalid config -> Network message without MessageType: Network message without MessageType 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'MessageType'
    Path:     NetworkMessage.schema.json#/required
"
`;

exports[`NetworkMessage schema (4) match fails for invalid config -> Network message without PublisherId: Network message without PublisherId 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'PublisherId'
    Path:     NetworkMessage.schema.json#/required
"
`;
